core/src/main/java/org/infinispan/remoting/transport/impl/MapResponseCollector.java

```diff
@@ -23,36 +24,43 @@
  * @author Dan Berindei
  * @since 9.1
  */
-public class SyncMapResponseCollector extends ValidResponseCollector<Map<Address, Response>> {
+public class MapResponseCollector extends ValidResponseCollector<Map<Address, Response>> {
    private final HashMap<Address, Response> map;
    private final boolean ignoreLeavers;
```

Comment:
I think the boolean can be removed and have 2 implementation. this is my suggestion (ps. the same idea for the remaining response collectors!)

```
public abstract class MapResponseCollector extends ValidResponseCollector<Map<Address, Response>> {
   public static MapResponseCollector defaultCollector() {
      return new DefaultCollector(4);
   }

   public static MapResponseCollector ignoreLeaversCollector() {
      return new IgnoreLeaversCollector(4);
   }

   public static MapResponseCollector defaultCollector(int size) {
      return new DefaultCollector(size);
   }

   public static MapResponseCollector ignoreLeaversCollector(int size) {
      return new IgnoreLeaversCollector(size);
   }

...
   private static class DefaultCollector extends MapResponseCollector {

      private DefaultCollector(int expectedSize) {
         super(expectedSize);
      }

      @Override
      protected Map<Address, Response> addTargetNotFound(Address sender) {
         recordException(ResponseCollectors.remoteNodeSuspected(sender));
         return null;
      }
   }

   private static class IgnoreLeaversCollector extends MapResponseCollector {

      private IgnoreLeaversCollector(int expectedSize) {
         super(expectedSize);
      }

      @Override
      protected Map<Address, Response> addTargetNotFound(Address sender) {
         map.put(sender, CacheNotFoundResponse.INSTANCE);
         return null;
      }
   }
```

Reply:
Yeah, your proposal is probably better. 

I didn't want to make these collectors too fancy because they were supposed to be only a crutch until we start using custom collectors on each call site (and avoid adding a `CompletableFuture` callback to transform the responses into something else). But considering how long this PR has taken, worked on them a little more.

Reply:
can you elaborate your idea? I'm curious :)